### PR TITLE
Adapting ParallaxComponent constructors to other components

### PR DESCRIPTION
**Pull request by erickzanardo**
# Description

Making ParallaxComponent constructor be more in the standard followed by SpriteComponent and SpriteAnimationComponent allowing a better use of the onLoad method on the component

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] This branch is based on the latest `master`
- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [x] I have made corresponding changes to the documentation
- [x] I have added examples for new features in `doc/examples`
- [x] The continuous integration (CI) is passing

### CODE REVIEW

**Review by spydon — APPROVED** [review]
Lgtm

File: doc/components.md [hunk 1]
```diff
@@ -199,11 +199,29 @@ The simplest `ParallaxComponent` is created like this:
 ```dart
 @override
 Future<void> onLoad() async {
-  final parallaxComponent = await loadParallaxComponent([bg.png, trees.png]);
+  final parallaxComponent = await loadParallaxComponent(['bg.png', 'trees.png']);
   add(parallax);
 }
 ```
 
+A ParallaxComponent can also "load itself" by implmenting the its own load method:
```
Comment (wolfenrain) [inline doc/components.md:207]:
```suggestion
A ParallaxComponent can also "load itself" by implementing the `onLoad` method:
```

File: doc/components.md [hunk 2]
```diff
@@ -263,7 +281,7 @@ The `Parallax` file contains an extension of the game which adds `loadParallax`,
 and `loadParallaxImage` so that it automatically uses your game's image cache instead of the global
 one. The same for the `ParallaxComponent` file, but that provides `loadParallaxComponent`.
 
-Two examples implementation can be found in the
+Three examples implementation can be found in the
```
Comment (wolfenrain) [inline doc/components.md:284]:
```suggestion
Three example implementations can be found in the
```

Also why is the link to the directory on a different line?

**Review by wolfenrain — APPROVED** [review]
LGTM, few small nits.

**Review by wolfenrain — APPROVED** [review]

**Review by renancaraujo — APPROVED** [review]

File: lib/src/components/parallax_component.dart [hunk 3]
```diff
@@ -34,30 +34,40 @@ extension ParallaxComponentExtension on Game {
 /// A full parallax, several layers of images drawn out on the screen and each
```
Comment (luanpotter) [inline lib/src/components/parallax_component.dart:34]: why we need `loadParallaxComponent` extension on Game? can't you just do ParallaxComponent.load? why a function just to call another one, w/o any difference? not sure I understand that

Comment (erickzanardo) [reply lib/src/components/parallax_component.dart:34]: That is something that we have agreed sometime now, when we did the sprite/animation refactor, we have the global cache (Flame.images) and the cache on the game class, all static load functions receive an instance of Images, with a default to the global cache.

This extension enforces the use of the game cache, so all assets loaded by functions like that are guaranteed to have used the game cache.

Comment (luanpotter) [reply lib/src/components/parallax_component.dart:34]: oh I see the images: images now

**Review by luanpotter — APPROVED** [review]
LGTM

**Pull request merged**